lib/internal/modules/esm/register.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const { getHooksProxy } = require('internal/modules/esm/loader');
+
+/**
+ * Register a single loader programmatically.
+ * @param {string} specifier
+ * @returns {Promise<void>}
+ * @example
+ * ```js
+ * await registerLoader('ts-node/esm')
+ * await registerLoader('./myLoader.js')
+ * ```
+ */
+async function registerLoader(specifier) {
+  const hooksProxy = getHooksProxy();
+
+  await hooksProxy.makeAsyncRequest('registerLoader', specifier)
```

Comment:
I think we have consensus for this to be sync.

@aduh95 @GeoffreyBooth @targos could you 👍👎 this comment for accent/decent?

Reply:
Nit: I think we can just name the message type "register" (this is all loaders, so suffixing with "loader" seems redundant/overly-verbose). If we need to be more specific in future, this is implementation detail transparent to users, so we can change it whenever.